src/rules/converters/space-before-function-paren.ts

```diff
@@ -1,29 +1,32 @@
 import { RuleConverter } from "../converter";
 
-const NOT_SUPPORTED_OPTIONS = ["constructor", "method"];
-
-function isExistedESLintOption(rule: string) {
-    return !NOT_SUPPORTED_OPTIONS.includes(rule);
-}
+const SUPPORTED_OPTIONS: string[] = ["anonymous", "asyncArrow", "named"];
 
 export const convertSpaceBeforeFunctionParen: RuleConverter = tslintRule => {
-    const ruleArguments = tslintRule.ruleArguments.filter(isExistedESLintOption);
-    const notices = NOT_SUPPORTED_OPTIONS.reduce<string[]>((acc, option) => {
-        if (tslintRule.ruleArguments.includes(option)) {
-            acc.push(`Option "${option}" is not supported by ESLint.`);
-        }
-        return acc;
-    }, []);
-
-    return {
-        rules: [
-            {
-                ...(tslintRule.ruleArguments.length !== 0 && {
-                    ruleArguments,
+    const argument: "always" | "never" | { [key: string]: "always" | "never" } | undefined =
+        tslintRule.ruleArguments[0];
+    const ruleName = "space-before-function-paren";
+    if (argument === "always" || argument === "never") {
+        return { rules: [{ ruleArguments: [argument], ruleName }] };
+    } else if (typeof argument === "object" && argument !== null) {
```

Comment:
The contents in the body of this `else if` are a little substantial to read through. Would you mind splitting them into a separate function? That'd be helpful for readability IMO.

Reply:
I'm sorry. I can't read English well. In this what you mean?

```typescript
const isObject = (o: unknown): o is Record<string, "always" | "never"> =>
    typeof o === "object" && o !== null;

// ...
} else if (isObject(argument)) {
// ...
```

Alternatively, I think it would be better to use `typeof argument === "undefined"` if `tslint.json` is valid.

```typescript
// ...
} else if (typeof argument === "undefined") {
    return { rules: [{ ruleName }] };
}
const notices = Object.keys(argument)
    .filter(option => !SUPPORTED_OPTIONS.includes(option))
    .map(option => `Option "${option}" is not supported by ESLint.`);
const filtered = Object.keys(argument)
    .filter(option => SUPPORTED_OPTIONS.includes(option))
    .reduce<Record<string, "always" | "never">>((obj, option) => {
        return { ...obj, [option]: argument[option] };
    }, {});
return {
    rules: [
        {
            ...(notices.length !== 0 && { notices }),
            ...(Object.keys(filtered).length !== 0 && { ruleArguments: [filtered] }),
            ruleName,
        },
    ],
};
```

What would you like to do specifically?

Reply:
Oh sorry for the ambiguity in my comments! Yes, your second suggestion around using `typeof` seems great. 

My overall goal was to reduce the amount of logic placed inside the `else` statement, since that can be confusing to read through. Moving that stuff into a separate function can be helpful to split the code up so each individual portion is understandable.